### PR TITLE
Make render and asset middlewares more robust

### PR DESCRIPTION
**Pull request by dmbch**
Closes: #120

### CODE REVIEW

File: packages/webpack/mixin.core.js [hunk 1]
```diff
@@ -57,7 +57,6 @@ class WebpackMixin extends Mixin {
         logLevel: 'silent',
         publicPath: webpackBrowserConfig.output.publicPath,
         watchOptions: webpackBrowserConfig.watchOptions,
-        serverSideRender: true,
```
Comment (ZauberNerd) [inline packages/webpack/mixin.core.js]: What does this change do?

File: packages/webpack/lib/middleware/render.js [hunk 2]
```diff
@@ -1,41 +1,79 @@
...
-    } else if (stats.hasErrors() || stats.hasWarnings()) {
-      const { errors, warnings } = stats.toJson();
-      resolvable.reject(
-        new Error(`Can't compile:\n${[...errors, ...warnings].join('\n')}`)
-      );
+      emitResult('error', compileError);
+    } else if (stats.hasErrors()) {
+      const { errors } = stats.toJson();
+      const details = errors
+        .map((str) => str.replace(/\033?\[[0-9]{1,2}m/g, ''))
```
Comment (ZauberNerd) [inline packages/webpack/lib/middleware/render.js]: What are you doing here? Replacing color codes?

Comment (ZauberNerd) [reply packages/webpack/lib/middleware/render.js]: Seems like you can set `colors: false` to the `stats.toJson()` method - https://webpack.js.org/configuration/stats/

Comment (dmbch) [reply packages/webpack/lib/middleware/render.js]: Yes, stripping color codes - I tried just using that setting, but it does not work the way we both thought. ;)

File: packages/webpack/mixin.core.js (hunk 1 above)
Comment (dmbch) [reply packages/webpack/mixin.core.js]: That setting instructs `webpack-dev-middleware` to make sure it sets the stats to `res.locals.webpackStats` before invoking the next middleware, allowing a developer to render the page body and manage the response to clients. See: https://github.com/webpack/webpack-dev-middleware#server-side-rendering

We do not use that feature any more - and it is marked as `experimental`, anyway.

File: packages/webpack/lib/utils/resolvable.js [hunk 3]
```diff
@@ -17,28 +19,33 @@ exports.Resolvable = class Resolvable {
...
+            (error) => void reject(error) || this.reject(error),
+            (executor) => this.reset(executor)
+          )
+        ).catch(this.reject);
+      }
+    };
+    this.resolve = (value) => {
+      Promise.resolve(value).then(
+        (value) => queue.slice().forEach((fn) => fn(null, value)),
+        this.reject
```
Comment (jhiode) [inline packages/webpack/lib/utils/resolvable.js]: Why do we need this additional promise? And this `reject` function will never be called as the promise is already resolved.

Comment (dmbch) [reply packages/webpack/lib/utils/resolvable.js]: I am afraid we do: `value` could be a `Promise`. That is the reason I added `this.reset` here, too.

Generally, I want this thing to mimick the behavior of a real `Promise`.

**Pull request merged**